src/TouchBarController.m

```diff
@@ -23,10 +26,22 @@ + (id)controllerWithNibNamed:(NSString *)name
     if (![[NSBundle mainBundle]
         loadNibNamed:name owner:controller topLevelObjects:&objects])
         return nil;
+    
+    DFRSystemModalShowsCloseBoxWhenFrontMost(YES);
+    [controller setPlacement:1];
 
     return controller;
 }
 
+- (id)init
+{
+    if (self = [super init]) {
+        _button = [[[NSCustomTouchBarItem alloc] initWithIdentifier:kControlButtonIdentifier] autorelease];
```

Comment:
This project does not use ARC, so this code will `autorelease` the `_button`, which could end up as a pointer to free memory.

Also the `_button` should be freed in `dealloc`.